mongo/src/main/java/org/axonframework/mongo/eventsourcing/tokenstore/MongoTokenStore.java

```diff
@@ -34,68 +39,73 @@
 import java.time.Duration;
 import java.time.Instant;
 import java.time.temporal.TemporalAmount;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
+import javax.annotation.PostConstruct;
 
 import static com.mongodb.client.model.Filters.*;
 import static com.mongodb.client.model.Projections.*;
 import static com.mongodb.client.model.Sorts.ascending;
 import static com.mongodb.client.model.Updates.combine;
 import static com.mongodb.client.model.Updates.set;
 import static java.lang.String.format;
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+import static org.axonframework.common.BuilderUtils.assertThat;
 
 /**
  * An implementation of TokenStore that allows you store and retrieve tracking tokens with MongoDB.
+ *
+ * @author Joris van der Kallen
+ * @since 3.1
  */
 public class MongoTokenStore implements TokenStore {
 
-    private final static Clock clock = Clock.systemUTC();
     private final static Logger logger = LoggerFactory.getLogger(MongoTokenStore.class);
+    private final static Clock clock = Clock.systemUTC();
+
     private final MongoTemplate mongoTemplate;
     private final Serializer serializer;
     private final TemporalAmount claimTimeout;
     private final String nodeId;
     private final Class<?> contentType;
 
     /**
-     * Creates a MongoTokenStore with a default claim timeout of 10 seconds, a default owner identifier and a default
-     * content type.
+     * Instantiate a {@link MongoTokenStore} based on the fields contained in the {@link Builder}.
+     * <p>
+     * Will assert that the {@link MongoTemplate} and {@link Serializer} are not {@code null}, and will throw an
+     * {@link AxonConfigurationException} if any of them is {@code null}.
      *
-     * @param mongoTemplate used to access the collection in which tracking tokens are stored
-     * @param serializer    serializer used to serialize tracking tokens
+     * @param builder the {@link Builder} used to instantiate a {@link MongoTokenStore} instance
      */
-    public MongoTokenStore(MongoTemplate mongoTemplate, Serializer serializer) {
-        this(mongoTemplate,
-             serializer,
-             Duration.ofSeconds(10),
-             ManagementFactory.getRuntimeMXBean().getName(),
-             byte[].class);
+    protected MongoTokenStore(Builder builder) {
+        builder.validate();
+        this.mongoTemplate = builder.mongoTemplate;
+        this.serializer = builder.serializer;
+        this.claimTimeout = builder.claimTimeout;
+        this.nodeId = builder.nodeId;
+        this.contentType = builder.contentType;
     }
 
     /**
-     * Creates a MongoTokenStore by using the given values.
+     * Instantiate a Builder to be able to create a {@link MongoTokenStore}.
+     * <p>
+     * The {@code claimTimeout} is defaulted to a 10 seconds duration, {@code nodeId} is defaulted to the
```

Comment:
Is the time unit of the `cleamTimeout` seconds? or milliseconds? Maybe good to note